dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageProblem.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Multimaps;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * A linkage error.
+ *
+ * @see <a href="https://jlbp.dev/glossary.html#linkage-error">Java Dependency Glossary: Linkage
+ *     Error</a>
+ */
+public final class LinkageProblem {
+
+  private final ErrorType errorType;
+  private final Symbol symbol;
+  private final ClassFile containingClass;
+  private final ClassFile sourceClass;
+
+  @VisibleForTesting
+  public LinkageProblem(
```

Comment:
I guess I'm missing something but I don't see anywhere a linkageproblem is created except in test code

Reply:
so where/how does LinkageChecker create them? I don't see that in this PR.

Reply:
The large diff in LinkageChecker.java is not rendered by default. Find 'LinkageChecker.java' and expand the diff before searching for "new LinkageProblem":

<img width="996" alt="Screen Shot 2020-07-10 at 13 46 58" src="https://user-images.githubusercontent.com/28604/87183270-d6637c00-c2b3-11ea-9089-ae38bef53935.png">